src/active-model/model.cr

```diff
@@ -496,22 +462,51 @@ abstract class ActiveModel::Model
     {% end %}
   end
 
+  # Declare attributes in real model
   macro attribute(name, converter = nil, mass_assignment = true, persistence = true, **tags, &block)
+    # Declaring correct type of attribute
     {% resolved_type = name.type.resolve %}
     {% if resolved_type.nilable? %}
       {% type_signature = resolved_type %}
     {% else %}
       {% type_signature = "#{resolved_type} | Nil".id %}
     {% end %}
 
+    # Assign instance variable to correct type
+
+    @[JSON::Field(
+      presence: true,
+      {% if !persistence %}
+        ignore: true,
+      {% end %}
+      {% if !converter.nil? %}
+        converter: {{converter}}
+      {% end %}
+    )]
+    @[YAML::Field(
+      presence: true,
+      {% if !persistence %}
+        ignore: true,
+      {% end %}
+      {% if !converter.nil? %}
+        converter: {{converter}}
+      {% end %}
+    )]
     @{{name.var}} : {{type_signature.id}}
 
+    @[JSON::Field(ignore: true)]
+    @[YAML::Field(ignore: true)]
+    getter? {{name.var}}_present : Bool = false
+
     # Attribute default value
     def {{name.var.id}}_default : {{ name.type }}
```

Comment:
Perhaps ignore this whole method if there is no default? This will prevent a potential runtime error.